### DISCUSSION TITLE
Manipulating math.equation.body using show

### DISCUSSION THREAD

**Discussion by tianze0926** [typst/typst#2242] [Q&A] [answered]
I am trying to add `display` style to all inline equations. Take the following inline equation for example:

```typst
$x = e/100$
```

To make it be of the same size as block equations:

```typst
$display(x = e/100)$
```

I want all inline quations to have `display` style, hence I tried the `show` command:
```typst
#show math.equation.where(block: false): it => ...
```

But I cannot get it working since however I tried to access `it.body` the compiler would crash.

It also confuses me that when the cursor hovers on `it` in the web app, the following shows up:

```
equation(
  body: sequence(
    children: ([x], [ ], [=], [ ], frac(num: [e], denom: [100])),
  ),
  block: false,
  numbering: none,
  supplement: [Equation],
),
equation(
  body: math-style(
    body: sequence(
      children: ([x], [ ], [=], [ ], frac(num: [e], denom: [100])),
    ),
    size: "display",
    cramped: false,
  ),
  block: false,
  numbering: none,
  supplement: [Equation],
),
```

I could not find any information about `sequence` and `math-style`. What are those and how can I manipulate them?

**Comment by frozolotl** [comment, marked as answer]
When you use the following code, the compiler infinitely recurses: whenever it sees an inline equation, it runs the show rule, which creates a new inline equation, the same show rule is run, and so on.
```typ
#show math.equation.where(block: false): it => $display(it)$
```

Until that behavior is improved, you can use the following workaround:
```typ
#show math.equation.where(block: false): it => {
  if it.has("label") and it.label == <stop-equation-recursion> {
    it
  } else {
    [$display(it)$ <stop-equation-recursion>]
  }
}
```

As to your other question, `sequence` and `math-style` are largely internal types that are instantiated in certain circumstances.
For one, `sequence` is just that, a sequence of individual elements which which are rendered consecutively. The following are all sequences: `hello *world*`, `$x y$`, `$x$ is not $y$`.
Then `math-style` is initialized when you want to style something in math mode. For instance: `$display(x)$`, `$inline(x)$`, `bold(x)`, `bb(x)`.

**Reply by xkevio** [reply]
This could also work I think.

```typ
#show math.equation.where(block: false): eq => {
  if eq.body.has("size") and eq.body.size == "display" {
    eq
  } else {
    $display(eq)$
  }
}
```

**Reply by tianze0926** [reply]
Big thanks for your quick and detailed reply!
